prow/crier/reporters/gcs/reporter.go

```diff
@@ -138,12 +138,11 @@ func (gr *gcsReporter) reportStartedJob(ctx context.Context, log *logrus.Entry,
 		return fmt.Errorf("failed to marshal started metadata: %w", err)
 	}
 
-	// Overwrite if it was uploaded by crier and there might be something new
-	var opts []io.WriterOptions
-	if existing {
-		opts = append(opts, io.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(false)})
-	}
-	return io.WriteContent(ctx, log, gr.opener, providers.GCSStoragePath(bucketName, path.Join(dir, prowv1.StartedStatusFile)), output, opts...)
+	// When come here and started.json already exist, it has to be uploaded by
+	// crier and expect an update. Otherwise uploading is fine.
+	// PreconditionDoesNotExist:true means create only when file not exist.
+	overwriteOpt := io.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(!existing)}
+	return io.WriteContent(ctx, log, gr.opener, providers.GCSStoragePath(bucketName, path.Join(dir, prowv1.StartedStatusFile)), output, overwriteOpt)
```

Comment:
How about just revert to the old version? I have a hard time understanding the line

`overwriteOpt := io.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(!existing)}`

whereas the old code of `if existing { ... }` is simpler to read for me. The old comment `Overwrite if it was uploaded by crier and there might be something new` is easier to understand as well.

Reply:
Or, as an alternative, it would be easier to read if we named two variables and used those instead. Something like:

```
overwrite := io.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(false)}
doNotOverwrite := io.WriterOptions{PreconditionDoesNotExist: utilpointer.BoolPtr(true)}
```

and then use these variables instead.

Reply:
I can see your point of confusion, myself was confused as well, nice thing is that this is guarded by unit tests that I added in previous PR.

In terms of readability, I'm not sure that I like to add these two global vars. Updated the comment and and one more var, hopefully this looks better